fix: hash token before persisting it

dappelt opened the pull request.

Commit:
fix: hash token before persisting it

emschwartz commented on index.js:
```diff
@@ -119,12 +119,12 @@ class Plugin extends AbstractBtpPlugin {
+            if (storedToken && storedToken !== tokenToAccount(token)) {
               throw new Error('incorrect token for account.' +
                 ' account=' + account +
                 ' token=' + token)
```
Wouldn't it be impossible to get here if the account is stored instead of the token? If you only store the account, you'll either get a new account or one that already exists, but there's no way to have an invalid credential for an account.

dappelt replied:
I added tests. Do you still have doubts that the code works as intended?

Commit:
test: add tests for BTP authentication

Commit:
refactor: move btp auth code into helper function

Commit:
chore: switch to standard for linting

Commit:
ci: enable lint and unit tests

Commit:
feat: hash tokens of existing accounts

the BTP token was previously written to the store in plaintext. This feature will hash the tokens of

existing accounts and delete the unhashed token from the store.

sharafian commented on src/token.js:
```diff
@@ -0,0 +1,64 @@
+            this._hashedToken === otherToken._hashedToken
+  }
+
+  exists () {
```
I think this should actually read from the store, otherwise [this line](https://github.com/interledgerjs/ilp-plugin-mini-accounts/pull/26/files/bdf3efc5f0fbbcbe65be9428e34d7916a6f07b32..f6800892f34ec4270a5c98db5a3a28154fe01232#diff-168726dbe96b3ce427e7fedce31bb0bcR123) could return false even if the `receivedToken.save` call had gone through from another login on the same account

dappelt replied:
good point

Commit:
fix: when checking if a token exists, read the token from the store

sharafian commented:
Can we test the race condition to make sure it was removed?

Commit:
test: does not race when storing the token

dappelt commented:
Added a test
